jina/types/document/__init__.py

```diff
@@ -531,7 +531,13 @@ def embedding(self) -> 'SparseArrayType':
 
         :return: the embedding of this Document
         """
-        return NdArray(self._pb_body.embedding).value
+        if self._pb_body.embedding.HasField('dense'):
+            dense_proto = self._pb_body.embedding.dense
+            return np.frombuffer(dense_proto.buffer, dtype=dense_proto.dtype).reshape(
+                dense_proto.shape
+            )
```

Comment:
this is simply `DenseNdArray(self._pb_body.embedding).value`

Reply:
> tend to reject, as this basically bypasses the `NdArray` class we implemented, sacrificing all quantization and corner case coverage, i.e. use simpler implementation to trade for faster speed.
> 
> I believe the key is to make `NdArray` class faster as we would still need it in advance usecase.

I see, Maybe we can do the same using`NdArray` but instead of creating an instance every time we can make it so that a `staticmethod` returns the numpy array. Will investigate further.